src/WingetCreateCore/Common/GitHub.cs

```diff
@@ -366,6 +369,24 @@ private async Task<PullRequest> SubmitPRAsync(string packageId, string version,
             }
         }
 
+        /// <summary>
+        /// Checks if the provided forked repository is behind on upstream commits and updates the default branch with the fetched commits. Update can only be a fast-forward update.
+        /// </summary>
+        /// <param name="forkedRepo"><see cref="Repository"/>Forked repository to be updated.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        private async Task UpdateForkedRepoWithUpstreamCommits(Repository forkedRepo)
+        {
+            var upstream = forkedRepo.Parent;
+            var compareResult = await this.github.Repository.Commit.Compare(upstream.Id, upstream.DefaultBranch, $"{forkedRepo.Owner.Login}:{forkedRepo.DefaultBranch}");
+
+            // Check to ensure that the update is only a fast-forward update.

```

Comment:
>// Check to ensure that the update is only a fast-forward update. [](http://example.com/codeflow?start=12&length=65)

suggest: for cases that cannot be ff'ed, throw a custom exception so the caller can catch it and display a corresponding error message

Reply:
Added custom exception to display the correct error message to the user.